brax/training/types.py

```diff
@@ -79,7 +79,7 @@ class NetworkFactory(Protocol[NetworkType]):
 
   def __call__(
       self,
-      observation_size: int,
+      observation_size: Union[int, Mapping[str, Tuple[int, ...]]],
```

Comment:
should this be Union[Tuple[int, ...], int] in the Mapping?

Reply:
Yes there seems to be no reason to enforce having <=1D obs be tuple-sized

Reply:
In the case of a dict, observation_size will always be Mapping[str, Tuple[int, ...]] the way I calculate it using tree_map, but in general having non-tuple sizes makes sense.